File: RobocodeLabo/src/laboratorio/FelipeVerdugoRobot.java
package laboratorio;

import robocode.JuniorRobot;

public class FelipeVerdugoRobot extends JuniorRobot {
	private StrategyJuniorRobot strategy;

	@Override
	public void run() {
		this.setStrategy(new NoHitStrategy());
		setColors(black, black, black, black, black);
		strategy.run(this);

	}

	public StrategyJuniorRobot getStrategy() {
		return strategy;
	}

	public void setStrategy(StrategyJuniorRobot strategy) {
		this.strategy = strategy;
	}

	/**
	 * onScannedRobot: What to do when you see another robot
	 */
	@Override
	public void onScannedRobot() {
		strategy.onScannedRobot(this);
	}

	/**
	 * onHitByBullet: What to do when you're hit by a bullet
	 */
	@Override
	public void onHitByBullet() {
		strategy.onHitByBullet(this);
	}

	/**
	 * onHitWall: What to do when you hit a wall
	 */
	@Override
	public void onHitWall() {
		strategy.onHitWall(this);
	}
}

File: Entrega-1-ko/src/laboratorio/FelipeVerdugoRobot.kt
package laboratorio

import robocode.JuniorRobot

class FelipeVerdugoRobot : JuniorRobot() {
    private var strategy: StrategyJuniorRobot = NoHitStrategy(true)

    override fun run() {
        setColors(black, black, black, black, black)
        strategy.run(this)
    }

    /**
     * onScannedRobot: What to do when you see another robot
     */
    override fun onScannedRobot() {
        strategy.onScannedRobot(this)
    }

    /**
     * onHitByBullet: What to do when you're hit by a bullet
     */
    override fun onHitByBullet() {
        strategy.onHitByBullet(this)
    }

    /**
     * onHitWall: What to do when you hit a wall
     */
    override fun onHitWall() {
        strategy.onHitWall(this)
    }
}